rotkehlchen/api/rest.py

```diff
@@ -5278,15 +5278,21 @@ def get_historical_prices_per_asset(
             interval: int,
             to_timestamp: Timestamp,
             from_timestamp: Timestamp,
+            exclude_timestamps: set[Timestamp],
             only_cache_period: int | None = None,
     ) -> dict[str, Any]:
         prices = {}
         no_prices_ts: list[Timestamp] = []
+        rate_limited_prices_ts: list[Timestamp] = []
         with (db := self.rotkehlchen.data.db).conn.read_ctx() as cursor:
             main_currency = db.get_setting(cursor=cursor, name='main_currency')
 
-        total_intervals = (to_timestamp - from_timestamp) // interval + 1
-        timestamps = [Timestamp(from_timestamp + (i * interval)) for i in range(total_intervals)]
+        # Generate timestamps for each interval between from_timestamp and to_timestamp
+        # Filter out any timestamps that are in the exclude_timestamps or in the future
+        timestamps = [
+            ts for i in range((to_timestamp - from_timestamp) // interval + 1)
+            if (ts := Timestamp(from_timestamp + (i * interval))) not in exclude_timestamps and ts <= ts_now()  # noqa: E501
+        ]
```

Comment:
mmmm here I'm not sure if it is calling ts_now() for each time that you check in the loop. Move it outside the loop. 

Also instead of checkign ts <= ts_now you can do 
```suggestion
        timestamps = [
            ts for i in range((min(ts_now(), to_timestamp) - from_timestamp) // interval + 1)
            if (ts := Timestamp(from_timestamp + (i * interval))) not in exclude_timestamps and ts <= ts_now()  # noqa: E501
        ]
```